From 076b352d19191bbe849bd24172ab538e93f9c89f Mon Sep 17 00:00:00 2001
From: Alex Kiernan <alex.kiernan@gmail.com>
Date: Sat, 30 Dec 2023 15:15:40 +0000
Subject: [PATCH 2/2] Ignore blank
 RUSTC_DEBUGINFO_MAP/RUSTC_CARGO_REGISTRY_SRC_TO_REMAP

If RUSTC_DEBUGINFO_MAP or RUSTC_CARGO_REGISTRY_SRC_TO_REMAP are empty,
avoid inserting `--remap-path-prefix` with no associated argument.

Signed-off-by: Alex Kiernan <alex.kiernan@gmail.com>
Upstream-Status: Submitted [https://github.com/rust-lang/rust/pull/119445]
---
 src/bootstrap/src/bin/rustc.rs | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/bootstrap/src/bin/rustc.rs b/src/bootstrap/src/bin/rustc.rs
index 070a2da6afb8..0c2c5863f5d7 100644
--- a/src/bootstrap/src/bin/rustc.rs
+++ b/src/bootstrap/src/bin/rustc.rs
@@ -125,13 +125,17 @@ fn main() {
     }
 
     if let Ok(map) = env::var("RUSTC_DEBUGINFO_MAP") {
-        cmd.arg("--remap-path-prefix").arg(&map);
+        if !map.is_empty() {
+            cmd.arg("--remap-path-prefix").arg(&map);
+        }
     }
     // The remap flags for Cargo registry sources need to be passed after the remapping for the
     // Rust source code directory, to handle cases when $CARGO_HOME is inside the source directory.
     if let Ok(maps) = env::var("RUSTC_CARGO_REGISTRY_SRC_TO_REMAP") {
         for map in maps.split('\t') {
-            cmd.arg("--remap-path-prefix").arg(map);
+            if !map.is_empty() {
+                cmd.arg("--remap-path-prefix").arg(map);
+            }
         }
     }
 
-- 
2.39.0

